src/config.cpp

```diff
@@ -9,6 +9,17 @@
 
 namespace Generators {
 
+std::string NormalizeProviderName(std::string_view name) {
+  if (name == "qnn") {
+    return "QNN";
+  } else if (name == "webgpu") {
+    return "WebGPU";
+  } else if (name == "dml") {
+    return "DML";
+  }
+  return std::string(name);
+}
+
```

Comment:
```suggestion
// Fix casing of certain historical names to match current Onnxruntime names
std::string_view NormalizeProviderName(std::string_view name) {
  if (name == "qnn") {
    return "QNN";
  } else if (name == "webgpu") {
    return "WebGPU";
  } else if (name == "dml") {
    return "DML";
  }
  return name;  // Return name unchanged
}
```

Figured might as well avoid allocating a std::string every time since it seems to be unnecessary.